python/annotation_bulk_clean.py

```diff
@@ -0,0 +1,50 @@
+# Preliminaries
+import synapseclient
+from synapseclient import Table, Schema
+import argparse
+
+# login to synapse
+syn = synapseclient.login(silent=True)
+
+# Get arguments for table id to be changed and column to focus on
+parser = argparse.ArgumentParser(
+    description='Get synapse table ids, column, old annotation, and new annotation')
+parser.add_argument('annotation_table_id', type=str,
+                    help='Synapse Table id for the annotations to be changed')
+parser.add_argument('cv_table_id', type=str,
+                    help='Synapse Table id for controlled vocabulary to be referenced')
+parser.add_argument('column', type=str,
+                    help='Name of column in annotation table to reference, i.e. "assay"')
+
+args = parser.parse_args()
+
+annotation_table_id = args.annotation_table_id
+cv_table_id = args.cv_table_id
+column = args.column
+
+# Create query for Controlled Vocabulary table using column specified
+cv_query = (f"SELECT key, value, existing FROM {cv_table_id} "
+            f"WHERE ((key = ('{column}')))")
+
+# Query the CV table using tableQuery() and convert the results into a dataframe.
+cv_view = syn.tableQuery(cv_query).asDataFrame()
+
+# Iterate through publications using query
+annotations_query = (f"SELECT {column} FROM {annotation_table_id}")
+
+# Query the publications table using tableQuery() and convert the results into a dataframe
+annotations_view = syn.tableQuery(annotations_query).asDataFrame().fillna("")
+
+# Iterate through each row in annoations_view and create list of values
+for i, row in annotations_view.iterrows():
+    annots = row[column]
+    if isinstance(annots, str):
+        annots = [annots]
+    new_annots = [cv_view.loc[cv_view['existing'] == x, 'value'].values[0]
+                  if x in cv_view['existing'].values else x
+                  for x in annots]
+    annotations_view.at[i, column] = new_annots
```

Comment:
Before updating the annotations, it may be worth to add an `if` statement here first, where the clause is whether the table we are fixing is All Files or not.  Since All Files have their annot columns as STRING type (instead of STRINGLIST like the other tables), we don't want the annotation values to be in a list.  If we save the values as a list, the resulting annotation value will have square bracket(s) around it, e.g.

<img width="282" alt="Screen Shot 2022-02-08 at 10 03 05 PM" src="https://user-images.githubusercontent.com/9377970/153131467-52eb6fbc-e5d0-4444-9488-009415647db4.png">

To help alleviate this, we could try something like:

```python
    new_annots = [cv_view.loc[cv_view['existing'] == x, 'value'].values[0]
                  if x in cv_view['existing'].values else x
                  for x in annots]
    if (table is All Files):  ## <-- will need to update this line with code
        new_annots = ", ".join(new_annots)
    annotations_view.at[i, column] = new_annots
```